crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2332,10 +2339,10 @@ mod tests {
 
         let events = db.take_salsa_events();
 
-        assert_function_query_was_not_run::<infer_definition_types, _, _>(
+        assert_function_query_was_not_run(
             &db,
-            |ty| &ty.function,
-            &first_public_def(&db, a, "x"),
+            infer_definition_types,
```

Comment:
This ended up improving the usability of the API! Nice.

Reply:
Yeah I think so. String matching is still a bit annoying but we at least have tests that tell us when the string matching stops working.